src/controllers/donorController.ts

```diff
@@ -64,77 +63,43 @@ export const getDonations = (req: Request, res: Response) => {
-            const descriptionUrls: string[] = [];
-            const foodUrls: string[] = [];
+            let descriptionUrls: string[] = [];
+            let foodUrls: string[] = [];
 
             if (req.files.descriptionImage) {
                 const descriptionImages: UploadedFile[] = req.files.descriptionImage as UploadedFile[];
-
                 if (!Array.isArray(descriptionImages)) {
-                    const imgRequest = descriptionImages as UploadedFile;
-                    blobSVC.createBlockBlobFromText(containerName, imgRequest.name, imgRequest.data, (error: Error) => {
-                        if (error) {
-                            console.error(`Error in createBlockBlobFromText: ${error}`);
-                            return res.status(500).json({error: error.message});
-                        }
-                    });
-                    const url = 'https://umifeedsimageupload.blob.core.windows.net/' + containerName + '/' + imgRequest.name; 
-                    descriptionUrls.push(url);
-                }
-
-                for (let i = 0; i < descriptionImages.length; i++) {
-                    const imgRequest = descriptionImages[i] as UploadedFile;
-
-                    blobSVC.createBlockBlobFromText(containerName, imgRequest.name, imgRequest.data, (error: Error) => {
-                        if (error) {
-                            console.error(`Error in createBlockBlobFromText: ${error}`);
-                            return res.status(500).json({error: error.message});
-                        }
-                    });
-                    const url = 'https://umifeedsimageupload.blob.core.windows.net/' + containerName + '/' + imgRequest.name; 
-                    descriptionUrls.push(url);
+                    const image = descriptionImages as UploadedFile;
+                    descriptionUrls = uploadFile(image, res);
+                } else {
+                    descriptionUrls = uploadFiles(descriptionImages, res);
                 }
-            }
-
+            } 
+    
             if (req.files.foodImage) {
                 const foodImages = req.files.foodImage as UploadedFile[];
-
                 if (!Array.isArray(foodImages)) {
-                    const imgRequest = foodImages as UploadedFile;
-                    blobSVC.createBlockBlobFromText(containerName, imgRequest.name, imgRequest.data, (error: Error) => {
-                        if (error) {
-                            console.error(`Error in createBlockBlobFromText: ${error}`);
-                            return res.status(500).json({error: error.message});
-                        }
-                    });
-                    const url = 'https://umifeedsimageupload.blob.core.windows.net/' + containerName + '/' + imgRequest.name; 
-                    foodUrls.push(url);
+                    const image = foodImages as UploadedFile;
+                    foodUrls = uploadFile(image, res);
```

Comment:
Hmm yeah that makes sense, but I think I passed in res in case there was an error with Azure so it throws an error and response for that and I would assume if it returns a status 500 then that is what the endpoint will return, but I'm not sure.

Reply:
Oh, I just saw your bottom comment, that makes sense to just thrown an error if it fails.